azure-pipelines.yml

```diff
@@ -16,14 +16,54 @@ jobs:
   steps:
   - script: |
       printenv
-        echo hello
-        pwd
-        ls
+      pwd
+      ls
+      echo "agend build directory"
+      echo $(Agent.BuildDirectory)
     displayName: filesystem check
   - script: |
+      cd $(Agent.BuildDirectory)/s
+      git clone https://github.com/oci-labs/rmw_opendds.git
+    displayName: clone rmw_opendds
+  - script: |
+        cd $(Agent.BuildDirectory)
+        git clone https://github.com/adamsj-oci/rmw_build
+        vcs import s < rmw_build/ros2.repos
+        echo "ls"
+        ls
+        echo "ls s"
+        ls s/
+    displayName:  rmw build repo and dependencies
+  - script: |
+        cd $(Agent.BuildDirectory)
+        source /opt/OpenDDS/setenv.sh
+        source /opt/ros/eloquent/setup.bash
+        printenv|grep DDS
+        printenv|grep ROS
+        colcon build --symlink-install --packages-up-to rmw_opendds_cpp
+    displayName:  build up to rmw_opendds_cpp
+  - script: |
+        cd $(Agent.BuildDirectory)
+        source /opt/OpenDDS/setenv.sh
+        source /opt/ros/eloquent/setup.bash
+        source install/local_setup.bash
+        printenv|grep DDS
+        printenv|grep ROS
+        colcon build --symlink-install --packages-up-to rcl_interfaces examples_rclcpp_minimal_publisher examples_rclcpp_minimal_subscriber
```

Comment:
Probably should add minimal service and client to get the srv files to build. I did manage to get the system_tests to build on a branch that would come after a clean build of the minimal examples of service and client.